### PR TITLE
Close poll feature implemented

### PR DESCRIPTION
**Pull request by vivekananda-reddy**

### CODE REVIEW

File: controllers/PollController.ts [hunk 1]
```diff
@@ -59,6 +67,79 @@ export default class PollController implements PollControllerI {
             .then((poll: Poll) => res.json(poll));
     }
 
+    /**
+     * Retrieves all polls from the database and returns an array of polls.
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON arrays containing the tuit objects
```
Comment (angela81ku) [inline controllers/PollController.ts]: body formatted as JSON arrays containing the poll objects

File: controllers/PollController.ts [hunk 2]
```diff
@@ -59,6 +67,79 @@ export default class PollController implements PollControllerI {
...
+    getAllPolls = (req: Request, res: Response) => {
+        PollController.pollDao.getAllPolls()
+            .then((poll: Poll[]) => res.json(poll));
+    }
+
+    /**
+     * Retrieves a poll from the database by id and returns a poll.
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON arrays containing the tuit objects
```
Comment (angela81ku) [inline controllers/PollController.ts]: body formatted as JSON arrays containing the poll objects

File: daos/PollDao.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import PollDaoI from "../interfaces/PollDaoI";
 import Poll from "../models/Poll";
 import PollModel from "../mongoose/PollModel";
+import TuitModel from "../mongoose/TuitModel";
```
Comment (angela81ku) [inline daos/PollDao.ts]: unnecessary import

File: controllers/PollController.ts [hunk 4]
```diff
@@ -59,6 +67,79 @@ export default class PollController implements PollControllerI {
...
+     * Close a poll if the request is from the creator. Else ignore the request
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents a response to client with the status of close operation
+     */
+    closePoll = async(req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+
+        const poll = await PollController.pollDao.getPollById(req.params.pid);
+        if(poll != null) {
```
Comment (angela81ku) [inline controllers/PollController.ts:130]: Might want to check if this would work for invalid pollId

**Review by angela81ku — CHANGES_REQUESTED** [review]

File: controllers/PollController.ts (hunk 1 above)
Comment (vivekananda-reddy) [reply controllers/PollController.ts]: Updated

File: controllers/PollController.ts (hunk 2 above)
Comment (vivekananda-reddy) [reply controllers/PollController.ts]: Updated

File: daos/PollDao.ts (hunk 3 above)
Comment (vivekananda-reddy) [reply daos/PollDao.ts]: Updated

File: controllers/PollController.ts (hunk 4 above)
Comment (vivekananda-reddy) [reply controllers/PollController.ts:130]: Performed testing in postman, it is working fine

**Pull request merged**